### PR TITLE
New Parser support create view

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

File: fe/fe-core/src/test/java/com/starrocks/analysis/AlterViewStmtTest.java [hunk 1]
```diff
@@ -1,196 +0,0 @@
...
-import mockit.Mocked;
-import org.junit.Assert;
-import org.junit.Before;
-import org.junit.Test;
-
-import java.io.StringReader;
-import java.util.LinkedList;
-import java.util.List;
-
-public class AlterViewStmtTest {
```
Comment (HangyuanLiu) [inline fe/fe-core/src/test/java/com/starrocks/analysis/AlterViewStmtTest.java:54]: This test has been moved into plan.ViewTest

File: fe/fe-core/src/main/java/com/starrocks/analysis/ColWithComment.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
 import com.starrocks.common.AnalysisException;
 import com.starrocks.common.FeNameFormat;
 
-public class ColWithComment {
+public class ColWithComment implements ParseNode{
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/analysis/ColWithComment.java:28]: a space?

File: fe/fe-core/src/main/java/com/starrocks/analysis/TableName.java [hunk 3]
```diff
@@ -65,21 +66,25 @@ public void analyze(Analyzer analyzer) throws AnalysisException {
...
+                    ErrorReport.reportAnalysisException(ErrorCode.ERR_CLUSTER_NAME_NULL);
+                }
+                db = ClusterNamespace.getFullName(connectContext.getClusterName(), db);
             }
-            db = ClusterNamespace.getFullName(connectContext.getClusterName(), db);
-        }
 
-        if (Strings.isNullOrEmpty(tbl)) {
-            throw new AnalysisException("Table name is null");
+            if (Strings.isNullOrEmpty(tbl)) {
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/analysis/TableName.java:83]: Is there any more check for tableName? Illegal characters, whitespace in name ?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SQLLabelBuilder.java [hunk 4]
```diff
@@ -0,0 +1,522 @@
...
+import com.starrocks.sql.ast.ValuesRelation;
+import com.starrocks.sql.optimizer.base.SetQualifier;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toList;
+
+public class SQLLabelBuilder {
+    public static String toSQL(ParseNode expr) {
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SQLLabelBuilder.java:60]: What is it called SQLLabelBuilder? All the functionality is translate `ParseNode` to sql string?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SQLLabelBuilder.java [hunk 5]
```diff
@@ -0,0 +1,522 @@
...
+            }
+
+            sqlBuilder.append(visit(queryRelation));
+
+            if (queryRelation.hasOrderByClause()) {
+                List<OrderByElement> sortClause = queryRelation.getOrderBy();
+                sqlBuilder.append(" order by ");
+                for (int i = 0; i < sortClause.size(); ++i) {
+                    sqlBuilder.append(visit(sortClause.get(i).getExpr()));
+                    sqlBuilder.append((sortClause.get(i).getIsAsc()) ? " asc" : " desc");
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SQLLabelBuilder.java:84]: upper case ?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SQLLabelBuilder.java [hunk 6]
```diff
@@ -0,0 +1,522 @@
...
+        public String visitSubquery(Subquery node, Void context) {
+            return "(" + visit(node.getQueryBlock()) + ")";
+        }
+
+        public String visitSysVariableDesc(SysVariableDesc node, Void context) {
+            return visitExpression(node, context);
+        }
+
+        @Override
+        public String visitTimestampArithmeticExpr(TimestampArithmeticExpr node, Void context) {
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SQLLabelBuilder.java:462]: Why is it special?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ViewDefBuilder.java [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+                sqlBuilder.append("DISTINCT");
+            }
+
+            List<String> selectListString = new ArrayList<>();
+            for (int i = 0; i < selectList.getItems().size(); ++i) {
+                SelectListItem item = selectList.getItems().get(i);
+                if (item.isStar()) {
+                    List<Expr> outputExpression = SelectAnalyzer.expandStar(item, stmt.getRelation());
+                    for (Expr expr : outputExpression) {
+                        selectListString.add(visit(expr) + " AS `" + toSQL(expr) + "`");
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ViewDefBuilder.java:123]: So many codes like `` + name + ``, how about replace it with an explicit function call `backQuote` ?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ViewDefBuilder.java [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+            }
+            return sqlBuilder.toString();
+        }
+
+        @Override
+        public String visitTableFunction(TableFunctionRelation node, Void scope) {
+            StringBuilder sqlBuilder = new StringBuilder();
+
+            sqlBuilder.append(node.getFunctionName());
+            sqlBuilder.append("(");
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ViewDefBuilder.java:276]: emm. This class shares a lot of code with SQLLabelBuilder, how about extract some common pattern to base class?

File: fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AnalyzeSingleTest.java [hunk 9]
```diff
@@ -323,43 +323,43 @@ public void testColumnNames() {
...
 
         query = analyzeSuccess(
                 "select * from (select v1 as v, sum(v2) from t0 group by v1) a inner join (select v1 as v,v2 from t0 order by v3) b on a.v = b.v");
-        Assert.assertEquals("v,sum(`v2`),v,v2", String.join(",", query.getColumnOutputNames()));
+        Assert.assertEquals("v,sum(v2),v,v2", String.join(",", query.getColumnOutputNames()));
 
         query = analyzeSuccess(
                 "select * from (select v1 as v, sum(v2) from t0 group by v1) a inner join (select v1 as v,v2 from t0 order by v3) b on a.v = b.v");
-        Assert.assertEquals("v,sum(`v2`),v,v2", String.join(",", query.getColumnOutputNames()));
+        Assert.assertEquals("v,sum(v2),v,v2", String.join(",", query.getColumnOutputNames()));
```
Comment (mofeiatwork) [inline fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AnalyzeSingleTest.java:346]: Does it means, all back quotes are removed from the output?

**Review by murphyatwork — DISMISSED** [review]

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ViewDefBuilder.java (hunk 8 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ViewDefBuilder.java:276]: ok

**Review by Seaven — APPROVED** [review]

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Review by Youngwb — APPROVED** [review]

**Pull request merged**